sdk/tests/iot/adu/test_az_iot_adu.c

```diff
@@ -689,19 +655,22 @@ static void test_az_iot_adu_client_parse_service_properties_with_null_file_url_v
   assert_int_equal(az_json_reader_next_token(&reader), AZ_OK);
 
   assert_int_equal(
-      az_iot_adu_client_parse_service_properties(
-          &adu_client,
-          &reader,
-          az_span_create(scratch_buffer, sizeof(scratch_buffer)),
-          &request,
-          &remainder),
-      AZ_OK);
+      az_iot_adu_client_parse_service_properties(&adu_client, &reader, &request), AZ_OK);
 
   // Workflow
   assert_int_equal(request.workflow.action, workflow_action);
   assert_memory_equal(az_span_ptr(request.workflow.id), workflow_id, sizeof(workflow_id) - 1);
 
   // Update Manifest
+  int32_t out_size;
+  assert_int_equal(
```

Comment:
lets not to the in place here. Put in buffer for now until a possible az_span api instead of string/null terminated.